shm/ship.cpp

```diff
@@ -0,0 +1,65 @@
+#include "ship.hpp"
+
+#include <iostream>
+#include <string>
+
+Ship::Ship() : id_(-1){};
+
+Ship::Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id)
```

Comment:
replace all int't to uint32_t. Maybe in this class it is not necessary, but you can do that, there will be less conflicts during merge ;)